### ISSUE TITLE
BuildOrder: Terran Orbital Command

### ISSUE THREAD

**Issue by alkurbatov** [davechurchill/commandcenter#12]
Currently one can specify OrbitalCommand as an instruction of OpeningBuildOrder. That leads to fun results:
* The bot reserves 550 minerals for construction (this looks like sc2api bug because it tells exact mineral cost for the building, currently it is Command Center (400) + Morph Orbital Command (150)).
* The building itself is never constructed (probably because the bot don't know how to build expansions or because it can't select a proper placement) however the building itself is added into the building queue and the overall build process is slowed down due to large mineral reserve.

It seems that either request to build Orbital Command should be automagically split into 'Create Command Center + Morph to Orbital) or that the bot should restrict such commands in general.
The latter looks more natural to me. In this case one should explicitly request morphing of command center.

I believe the bot behaves the same for the zerg's buildings.

**Comment by davechurchill** [comment]
I will make the change that requesting an orbital command is a request to morph a CC into one, I just haven't incorporated ANY morphed buildings yet. I am currently waiting on the API to add a specific functionality related to upgrades / supply costs before I make these changes.

**Comment by alkurbatov** [comment]
Yep, I saw that the bot can't morph buildings and units. Just wanted to point at a corner case with Orbital Command.

**Comment by davechurchill** [comment]
I have a system I am working on to make arbitrary units and upgrades in a build order ( see build type and techtree) but I am waiting on Blizzard to add 2 pieces of data to the API (unit type supply and upgrade costs) before I can finish it.

**Comment by davechurchill** [comment]
Actually since you have been so helpful, would you mind taking a look over TechTree.cpp and seeing if I have made any errors? The file is so large it would be nice to have a 2nd set of eyes on it.

**Comment by davechurchill** [comment]
Please post here for tech tree: https://github.com/davechurchill/commandcenter/issues/15

**Comment by davechurchill** [comment]
Closing the issue and editing the other issue to include orbitals

**Comment by alkurbatov** [comment]
>Actually since you have been so helpful, would you mind taking a look over TechTree.cpp and seeing if I have made any errors? The file is so large it would be nice to have a 2nd set of eyes on it.

Ok, but it will take some time.

**Comment by davechurchill** [comment]
It will be an ongoing effort, so no problems. I also plan on adding the UnitTypeID -> AbilityID mapping within that structure, for example
